bids/layout/utils.py

```diff
@@ -0,0 +1,78 @@
+import json
+from pathlib import Path
+from packaging.version import Version
+
+
+# As per https://bids.neuroimaging.io/bids_spec1.1.1.pdf
+desc_fields = {
+    Version("1.1.1"): {
+        "required": ["Name", "BIDSVersion"],
+        "recommended": ["License"],
+        "optional": ["Authors", "Acknowledgements", "HowToAcknowledge",
+                     "Funding", "ReferencesAndLinks", "DatasetDOI"]
+    }
+}
+
+
+def get_description_fields(version, type_):
+    if isinstance(version, str):
+        version = Version(version)
+    if not isinstance(version, Version):
+        raise TypeError("Version must be a string or a packaging.version.Version object.")
+
+    if version in desc_fields:
+        return desc_fields[version][type_]
+    return desc_fields[max(desc_fields.keys())][type_]
+
+
+def write_derivative_description(source_dir, name, bids_version='1.1.1', exist_ok=False, **desc_kwargs):
+
+    """
+     Write a dataset_description.json file for a new derivative folder.
+       source_dir : Directory of the BIDS dataset that has been derived.
+                    This dataset can itself be a derivative.
+       name : Name of the derivative dataset.
+       bid_version: Version of the BIDS standard.
+       desc_kwargs: Dictionary of entries that should be added to the
+                    dataset_description.json file.
+       exist_ok : Control the behavior of pathlib.Path.mkdir when a derivative folder
+                  with this name already exists.
+    """
+    if source_dir is str:
```

Comment:
This will fail. You would want `isinstance(source_dir, str)`, but there's no real point. Just unconditionally set `source_dir = Path(source_dir)`. This will coerce any string or `os.PathLike` object.